packages/sdk/src/sound/setMediaStateOptions.ts

```diff
@@ -4,9 +4,9 @@
  */
 
 /**
- * Parameters to the `Sound.setSoundState` and `Actor.startSound` calls.
+ * Parameters to the `MediaInstance.setState`, `Actor.startSound`, and `Actor.startVideoStream` calls.
  */
-export type SetSoundStateOptions = {
+export type SetMediaStateOptions = {
```

Comment:
>SetMediaStateOptions [](start = 12, length = 20)

SetMediaStateOptions is becoming a confusion of different types. I'd like to see this factored out into base a base type and media-type-specific types:
```
export type SetMediaStateOptions = {
   // ...common fields
};

export type SetAudioStreamStateOptions = SetMediaStateOptions & {
   // ...audio-specific options
   doppler: number;
   pitch: number;
};

export type SetVideoStreamStateOptions = SetMediaStateOptions & {
   // ...video-specific options
};

Reply:
If there were 3 or more media types, I could see an argument for splitting it up. Considering there are only 2 types, and there are only 1 audio-specific option (Doppler) and 2 videostream-specific options (Visible and Looping), I think that the code complexity will be higher than the benefit.

---
In reply to: [301167926](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301167926) [](ancestors = 301167926)

Reply:
Animation will end up using this, correct? That makes three. Users will try to apply doppler to animations.

---
In reply to: [301348602](https://github.com/microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301348602) [](ancestors = 301348602,301167926)